src/modules/launcher/Wox.Plugin/Common/Win32/Win32Helpers.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Wox.Plugin.Common.Win32
+{
+    public static class Win32Helpers
+    {
+        /// <summary>
+        /// Detects the type of system firmware which is equal to the boot type by calling the method <see cref="NativeMethods.GetFirmwareType"/>.
+        /// </summary>
+        /// <returns>Firmware type like Uefi or Bios.</returns>
+        public static FirmwareType GetSystemFirmwareType()
+        {
+            FirmwareType firmwareType = default;
+            _ = NativeMethods.GetFirmwareType(ref firmwareType);
+            return firmwareType;
+        }
+
+        /// <summary>
+        /// Returns the last Win32 Error code thrown by a native method if enabled for this method.
+        /// </summary>
+        /// <returns>The error code as int value.</returns>
+        public static int GetLastError()
+        {
+            // should be changed to Marshal.GetLastPInvokeError() in .NET 6
+            return Marshal.GetLastWin32Error();
```

Comment:
Is your branch up tobdate with main? We are now on .Net 6.

Reply:
oh i wasn't up to date